python/lsst/ap/association/assoc_afw_utils.py

```diff
@@ -0,0 +1,279 @@
+              'fluxZeroPointErr': flux0_err}
+    return values
+
+
+def add_dia_source_aliases_to_catalog(source_catalog):
+    """Create the needed aliases on fluxes and other columns if the input
+    catalog does not already have them.
+
+    Parameters
+    ----------
+    source_catalog : `lsst.afw.table.SourceCatalog`
+        Input source catalog to add alias columns to.
+    """
+    if not source_catalog.getSchema().contains(make_minimal_dia_source_schema()):
+        # Create aliases to appropriate flux fields if they exist.
+        schema_names = source_catalog.getSchema().getNames()
+        if 'base_PsfFlux_flux' in schema_names and \
+           'base_PsfFlux_fluxSigma' in schema_names and \
+           'psFlux' not in schema_names and \
+           'psFluxErr' not in schema_names:
+                alias_map = source_catalog.getSchema().getAliasMap()
+                alias_map.set('psFlux', 'base_PsfFlux_flux')
+                alias_map.set('psFluxErr', 'base_PsfFlux_fluxSigma')
+
+
+def convert_dia_source_to_asssoc_schema_and_calibrate(dia_sources,
+                                                      obj_ids=None,
+                                                      exposure=None):
+    """Create aliases in for the input source catalog schema and overwrite
+    values if requested.
+
+    Parameters
+    ----------
+    dia_sources : `lsst.afw.table.SourceCatalog`
+        Input source catalog to alias and overwrite values of.
+    obj_ids:  array-like of `int`
+        DIAObject ids to associated with the input DIASources. Overwrites
+        the current value in diaObjectId column.
+    exposure : `lsst.afw.image.Exposure`
+        Exposure to copy flux and filter information from.
+    """
+
+    output_dia_sources = afwTable.SourceCatog(
+        make_minimal_dia_source_schema())
+    output_dia_sources.reserve(len(dia_sources))
+    add_dia_source_aliases_to_catalog(dia_sources)
+    dia_source_schema = dia_sources.getSchema()
+
+    if exposure is None:
+        exp_dict = None
+    else:
+        exp_dict = get_ccd_visit_info_from_exposure(exposure)
+
+    for src_idx, source_record in enumerate(dia_sources):
+        if obj_ids is None:
+            obj_id = None
+        else:
+            obj_id = obj_ids[src_idx]
+        overwrite_dict = make_overwrite_dict(source_record, obj_id, exp_dict)
+        output_record = output_dia_sources.addNew()
+        copy_source_record(source_record,
+                           output_record,
+                           dia_source_schema,
+                           overwrite_dict)
+    return output_dia_sources
+
+
+def make_overwrite_dict(source_record, obj_id=None, exp_dict=None):
+    """Create a dictionary of values to overwrite the values stored in an
+    afw.table or when storing values in a DB.
+
+    Parameters
+    ----------
+    source_record : `lsst.afw.table.SourceRecord`
+        Input source catalog to modify values of if needed.
```

Comment:
"if needed" implies that this is optional. Is it? What is the default value in that case?